lucene/core/src/java/org/apache/lucene/index/FieldInfos.java

```diff
@@ -63,21 +62,30 @@ public FieldInfos(FieldInfo[] infos) {
     boolean hasNorms = false;
     boolean hasDocValues = false;
     boolean hasPointValues = false;
-    
-    TreeMap<Integer, FieldInfo> byNumber = new TreeMap<>();
+
+    int size = 0; // number of elements in byNumberTemp
+    int capacity = 10; // byNumberTemp's capacity
```

Comment:
I think we can simplify a bit by removing these two local variables:
 - `capacity` can be replaced with byNumberTemp.length
 - `size` seems only useful to terminate early when creating a list of the field infos from byNumberTemp. But since ArrayUtil.grow is used, arrays are never more than ~1/8th larger than the number of slots that are actually used, so it doesn't buy much?

Reply:
@jpountz Thank you for the suggestion on `capacity` - very nice one.  For the `size`, later we use this variable to create the final array: `byNumber = new FieldInfo[size]`. We can also do: `byNumber = new FieldInfo[byNumberTemp.length]`, but then `byNumber` may have some empty slots.